File: impl/app/android/app/src/main/java/com/ak/contexta/ui/addword/AddWordScreen.kt
package com.ak.contexta.ui.addword

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardActions
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.CheckCircle
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.OutlinedTextFieldDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.ImeAction
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.ak.contexta.domain.model.WordSense
import com.ak.contexta.ui.components.AppButton
import com.ak.contexta.ui.components.AppButtonVariant
import com.ak.contexta.ui.components.AppCard
import com.ak.contexta.ui.components.AppTopBar
import com.ak.contexta.ui.components.LoadingIndicator
import com.ak.contexta.ui.theme.Background
import com.ak.contexta.ui.theme.BodyText
import com.ak.contexta.ui.theme.Error
import com.ak.contexta.ui.theme.Hairline
import com.ak.contexta.ui.theme.Ink
import com.ak.contexta.ui.theme.Muted
import com.ak.contexta.ui.theme.MutedSoft
import com.ak.contexta.ui.theme.PhoneticStyle
import com.ak.contexta.ui.theme.Primary
import com.ak.contexta.ui.theme.Success
import com.ak.contexta.ui.theme.SurfaceSoft

@Composable
fun AddWordScreen(
    onBack: () -> Unit,
    viewModel: AddWordViewModel = hiltViewModel()
) {
    val state by viewModel.state.collectAsStateWithLifecycle()

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Background)
    ) {
        AppTopBar(title = "录入单词", onBack = onBack)

        if (state.success != null) {
            AddWordResultContent(
                success = state.success!!,
                onAddAnother = { viewModel.reset() },
                onDone = onBack
            )
        } else {
            AddWordInputContent(
                state = state,
                onInputChange = viewModel::onInputChange,
                onSubmit = viewModel::submit,
                onRetry = viewModel::submit
            )
        }
    }
}

@Composable
private fun AddWordInputContent(
    state: AddWordUiState,
    onInputChange: (String) -> Unit,
    onSubmit: () -> Unit,
    onRetry: () -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(16.dp)
    ) {
        AppCard(modifier = Modifier.fillMaxWidth()) {
            Column(modifier = Modifier.fillMaxWidth()) {
                Text(
                    text = "输入英文单词",
                    style = MaterialTheme.typography.headlineMedium
                )
                Spacer(modifier = Modifier.height(12.dp))
                OutlinedTextField(
                    value = state.input,
                    onValueChange = onInputChange,
                    modifier = Modifier.fillMaxWidth(),
                    placeholder = { Text("例如：serendipity", color = MutedSoft) },
                    singleLine = true,
                    shape = RoundedCornerShape(8.dp),
                    keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done),
                    keyboardActions = KeyboardActions(onDone = { onSubmit() }),
                    colors = OutlinedTextFieldDefaults.colors(
                        focusedBorderColor = Primary,
                        unfocusedBorderColor = Hairline,
                        focusedContainerColor = Background,
                        unfocusedContainerColor = Background
                    )
                )
                Spacer(modifier = Modifier.height(16.dp))
                AppButton(
                    text = "生成释义并加入生词库",
                    onClick = onSubmit,
                    modifier = Modifier.fillMaxWidth(),
                    enabled = state.input.isNotBlank() && !state.isSubmitting
                )
                Spacer(modifier = Modifier.height(8.dp))
                Text(
                    text = "本地词库没有该词时，将调用 AI 生成音标、释义与例句",
                    style = MaterialTheme.typography.bodySmall,
                    color = Muted
                )
            }
        }

        state.invalidInput?.let { message ->
            Spacer(modifier = Modifier.height(12.dp))
            Text(
                text = message,
                style = MaterialTheme.typography.bodyMedium,
                color = Error
            )
        }

        state.error?.let { message ->
            Spacer(modifier = Modifier.height(12.dp))
            AppCard() {
                Column(modifier = Modifier.fillMaxWidth()) {
                    Text(
                        text = message,
                        style = MaterialTheme.typography.bodyMedium,
                        color = BodyText
                    )
                    Spacer(modifier = Modifier.height(12.dp))
                    AppButton(text = "重试", onClick = onRetry)
                }
            }
        }

        if (state.isSubmitting) {
            Spacer(modifier = Modifier.height(24.dp))
            LoadingIndicator(
                message = state.stageMessage ?: "处理中…",
                modifier = Modifier.fillMaxWidth()
            )
        }
    }
}

@Composable
private fun AddWordResultContent(
    success: AddWordSuccessData,
    onAddAnother: () -> Unit,
    onDone: () -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(16.dp)
    ) {
        // 加入状态徽标
        AppCard() {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 16.dp, vertical = 12.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                if (success.addedToVocabulary) {
                    Icon(
                        imageVector = Icons.Outlined.CheckCircle,
                        contentDescription = null,
                        tint = Success,
                        modifier = Modifier.size(20.dp)
                    )
                } else {
                    Text(
                        text = "•",
                        style = MaterialTheme.typography.titleMedium,
                        color = Muted
                    )
                }
                Spacer(modifier = Modifier.width(8.dp))
                Text(
                    text = when {
                        success.addedToVocabulary && success.alreadyExisted ->
                            "已加入生词库（复用本地词库释义）"
                        success.addedToVocabulary -> "已加入生词库"
                        else -> "该词已在生词库中"
                    },
                    style = MaterialTheme.typography.bodyMedium,
                    color = if (success.addedToVocabulary) Success else BodyText,
                    fontWeight = FontWeight.SemiBold
                )
            }
        }

        Spacer(modifier = Modifier.height(16.dp))

        // 单词详情卡片
        AppCard() {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(8.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Text(
                    text = success.word,
                    style = MaterialTheme.typography.headlineLarge
                )
                if (success.phonetic != null) {
                    Spacer(modifier = Modifier.height(4.dp))
                    Text(
                        text = success.phonetic,
                        style = PhoneticStyle
                    )
                }

                success.senses.forEach { sense ->
                    Spacer(modifier = Modifier.height(16.dp))
                    SenseBlock(sense)
                }
            }
        }

        Spacer(modifier = Modifier.height(24.dp))

        AppButton(
            text = "再录一个",
            onClick = onAddAnother,
            modifier = Modifier.fillMaxWidth()
        )
        Spacer(modifier = Modifier.height(8.dp))
        AppButton(
            text = "返回生词本",
            onClick = onDone,
            modifier = Modifier.fillMaxWidth(),
            variant = AppButtonVariant.Secondary
        )
    }
}

@Composable
private fun SenseBlock(sense: WordSense) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(10.dp))
            .background(SurfaceSoft)
            .padding(12.dp)
    ) {
        Row(verticalAlignment = Alignment.Top) {
            Text(
                text = sense.partOfSpeech,
                style = MaterialTheme.typography.labelLarge,
                color = Primary,
                modifier = Modifier.padding(top = 2.dp)
            )
            Spacer(modifier = Modifier.width(8.dp))
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = sense.chineseMeaning,
                    style = MaterialTheme.typography.titleSmall,
                    fontWeight = FontWeight.SemiBold,
                    color = Ink
                )
                Spacer(modifier = Modifier.height(2.dp))
                Text(
                    text = sense.englishDefinition,
                    style = MaterialTheme.typography.bodySmall,
                    color = Muted
                )
            }
        }

        sense.examples.forEach { example ->
            Spacer(modifier = Modifier.height(10.dp))
            Column {
                Text(
                    text = example.sentenceEn,
                    style = MaterialTheme.typography.bodySmall,
                    color = BodyText
                )
                Text(
                    text = example.sentenceZh,
                    style = MaterialTheme.typography.bodySmall,
                    color = MutedSoft
                )
            }
        }
    }
}

File: impl/app/flutter/lib/ui/addword/add_word_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../../core/components/app_button.dart';
import '../../core/components/app_card.dart';
import '../../core/components/app_top_bar.dart';
import '../../core/components/loading_indicator.dart';
import '../../core/theme/app_colors.dart';
import '../../core/theme/app_dimens.dart';
import '../../core/theme/app_type.dart';
import '../../domain/model/word_detail.dart';
import 'add_word_controller.dart';

/// AddWord 页（对照 Kotlin AddWordScreen.kt）：
/// - 顶栏「录入单词」+ 返回
/// - 输入态：AppCard 内「输入英文单词」标题 + OutlinedTextField +
///   「生成释义并加入生词库」按钮（空白/提交中禁用）+ 说明文案；
///   invalidInput 红字 / error 卡片（重试按钮）；提交中 LoadingIndicator
///   （阶段进度文案）
/// - 结果态：加入状态徽标（✓ 已加入生词库 / • 该词已在生词库中）+
///   单词详情卡（词头 22sp + 音标 + SenseBlock 词性块）+
///   「再录一个」+「返回生词本」
class AddWordScreen extends ConsumerStatefulWidget {
  const AddWordScreen({super.key, required this.onBack});

  final VoidCallback onBack;

  @override
  ConsumerState<AddWordScreen> createState() => _AddWordScreenState();
}

class _AddWordScreenState extends ConsumerState<AddWordScreen> {
  final TextEditingController _inputController = TextEditingController();

  @override
  void dispose() {
    _inputController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final state = ref.watch(addWordControllerProvider);
    final controller = ref.read(addWordControllerProvider.notifier);

    return Scaffold(
      backgroundColor: AppColors.background,
      body: Column(
        children: [
          AppTopBar(title: '录入单词', onBack: widget.onBack),
          Expanded(
            child: state.success != null
                ? _AddWordResultContent(
                    success: state.success!,
                    onAddAnother: controller.reset,
                    onDone: widget.onBack,
                  )
                : _AddWordInputContent(
                    state: state,
                    controller: controller,
                    inputController: _inputController,
                  ),
          ),
        ],
      ),
    );
  }
}

/// 输入态内容（对照 Kotlin AddWordInputContent）。
class _AddWordInputContent extends StatelessWidget {
  const _AddWordInputContent({
    required this.state,
    required this.controller,
    required this.inputController,
  });

  final AddWordUiState state;
  final AddWordController controller;
  final TextEditingController inputController;

  @override
  Widget build(BuildContext context) {
    return ListView(
      padding: const EdgeInsets.all(AppSpacing.md),
      children: [
        AppCard(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.stretch,
            children: [
              Text(
                '输入英文单词',
                style: AppType.textTheme.headlineMedium,
              ),
              const SizedBox(height: AppSpacing.sm),
              TextField(
                controller: inputController,
                onChanged: controller.onInputChange,
                textInputAction: TextInputAction.done,
                onSubmitted: (_) => controller.submit(),
                inputFormatters: [
                  // 对照 Kotlin WORD_PATTERN 的前置过滤：输入时即拦截非法字符
                  FilteringTextInputFormatter.allow(
                    RegExp(r"[A-Za-z'\- ]"),
                  ),
                ],
                decoration: InputDecoration(
                  hintText: '例如：serendipity',
                  hintStyle: AppType.textTheme.bodyMedium
                      ?.copyWith(color: AppColors.mutedSoft),
                  enabledBorder: OutlineInputBorder(
                    borderRadius: BorderRadius.circular(AppRadius.sm),
                    borderSide: const BorderSide(color: AppColors.hairline),
                  ),
                  focusedBorder: OutlineInputBorder(
                    borderRadius: BorderRadius.circular(AppRadius.sm),
                    borderSide: const BorderSide(color: AppColors.primary),
                  ),
                  filled: true,
                  fillColor: AppColors.background,
                ),
                style: AppType.textTheme.bodyLarge,
              ),
              const SizedBox(height: AppSpacing.md),
              AppButton(
                text: '生成释义并加入生词库',
                onClick: controller.submit,
                enabled: state.input.trim().isNotEmpty && !state.isSubmitting,
              ),
              const SizedBox(height: AppSpacing.xs),
              Text(
                '本地词库没有该词时，将调用 AI 生成音标、释义与例句',
                style: AppType.textTheme.bodySmall
                    ?.copyWith(color: AppColors.muted),
              ),
            ],
          ),
        ),
        if (state.invalidInput != null) ...[
          const SizedBox(height: AppSpacing.sm),
          Text(
            state.invalidInput!,
            style: AppType.textTheme.bodyMedium
                ?.copyWith(color: AppColors.error),
          ),
        ],
        if (state.error != null) ...[
          const SizedBox(height: AppSpacing.sm),
          AppCard(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.stretch,
              children: [
                Text(
                  state.error!,
                  style: AppType.textTheme.bodyMedium
                      ?.copyWith(color: AppColors.bodyText),
                ),
                const SizedBox(height: AppSpacing.sm),
                Align(
                  alignment: Alignment.centerLeft,
                  child: AppButton(text: '重试', onClick: controller.submit),
                ),
              ],
            ),
          ),
        ],
        if (state.isSubmitting) ...[
          const SizedBox(height: AppSpacing.lg),
          LoadingIndicator(
            message: state.stageMessage ?? '处理中…',
          ),
        ],
      ],
    );
  }
}

/// 结果态内容（对照 Kotlin AddWordResultContent）：
/// 状态徽标 + 单词详情卡 + 再录一个 / 返回生词本。
class _AddWordResultContent extends StatelessWidget {
  const _AddWordResultContent({
    required this.success,
    required this.onAddAnother,
    required this.onDone,
  });

  final AddWordSuccessData success;
  final VoidCallback onAddAnother;
  final VoidCallback onDone;

  @override
  Widget build(BuildContext context) {
    return ListView(
      padding: const EdgeInsets.all(AppSpacing.md),
      children: [
        // 加入状态徽标
        AppCard(
          padding: const EdgeInsets.symmetric(
            horizontal: AppSpacing.md,
            vertical: AppSpacing.sm,
          ),
          child: Row(
            children: [
              if (success.addedToVocabulary)
                Icon(Icons.check_circle_outline,
                    size: 20, color: AppColors.success)
              else
                Text(
                  '•',
                  style: AppType.textTheme.titleMedium
                      ?.copyWith(color: AppColors.muted),
                ),
              const SizedBox(width: AppSpacing.xs),
              Expanded(
                child: Text(
                  switch ((success.addedToVocabulary, success.alreadyExisted)) {
                    (true, true) => '已加入生词库（复用本地词库释义）',
                    (true, false) => '已加入生词库',
                    (false, _) => '该词已在生词库中',
                  },
                  style: AppType.textTheme.bodyMedium?.copyWith(
                    color: success.addedToVocabulary
                        ? AppColors.success
                        : AppColors.bodyText,
                    fontWeight: FontWeight.w600,
                  ),
                ),
              ),
            ],
          ),
        ),
        const SizedBox(height: AppSpacing.md),
        // 单词详情卡
        AppCard(
          padding: const EdgeInsets.all(AppSpacing.xs),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.stretch,
            children: [
              Text(
                success.word,
                textAlign: TextAlign.center,
                style: AppType.textTheme.headlineLarge
                    ?.copyWith(color: AppColors.ink),
              ),
              if (success.phonetic != null) ...[
                const SizedBox(height: AppSpacing.xxs),
                Text(
                  success.phonetic!,
                  textAlign: TextAlign.center,
                  style: AppType.phonetic,
                ),
              ],
              for (final sense in success.senses) ...[
                const SizedBox(height: AppSpacing.md),
                _SenseBlock(sense: sense),
              ],
            ],
          ),
        ),
        const SizedBox(height: AppSpacing.lg),
        AppButton(text: '再录一个', onClick: onAddAnother),
        const SizedBox(height: AppSpacing.xs),
        AppButton(
          text: '返回生词本',
          onClick: onDone,
          variant: AppButtonVariant.secondary,
        ),
      ],
    );
  }
}

/// 词性块（对照 Kotlin AddWordScreen.SenseBlock：SurfaceSoft + 10dp 圆角）。
class _SenseBlock extends StatelessWidget {
  const _SenseBlock({required this.sense});

  final WordSense sense;

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: const EdgeInsets.all(AppSpacing.sm),
      decoration: BoxDecoration(
        color: AppColors.surfaceSoft,
        borderRadius: BorderRadius.circular(10),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Padding(
                padding: const EdgeInsets.only(top: 2),
                child: Text(
                  sense.partOfSpeech,
                  style: AppType.textTheme.labelLarge
                      ?.copyWith(color: AppColors.primary),
                ),
              ),
              const SizedBox(width: AppSpacing.xs),
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      sense.chineseMeaning,
                      style: AppType.textTheme.titleSmall
                          ?.copyWith(
                              fontWeight: FontWeight.w600,
                              color: AppColors.ink),
                    ),
                    const SizedBox(height: 2),
                    Text(
                      sense.englishDefinition,
                      style: AppType.textTheme.bodySmall
                          ?.copyWith(color: AppColors.muted),
                    ),
                  ],
                ),
              ),
            ],
          ),
          for (final example in sense.examples) ...[
            const SizedBox(height: 10),
            Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  example.sentenceEn,
                  style: AppType.textTheme.bodySmall
                      ?.copyWith(color: AppColors.bodyText),
                ),
                Text(
                  example.sentenceZh,
                  style: AppType.textTheme.bodySmall
                      ?.copyWith(color: AppColors.mutedSoft),
                ),
              ],
            ),
          ],
        ],
      ),
    );
  }
}
